engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/typeinformation/FlinkBaseTypeInfoRegister.scala

```diff
@@ -0,0 +1,45 @@
+package pl.touk.nussknacker.engine.flink.api.typeinformation
+
+import org.apache.flink.api.common.functions.InvalidTypesException
+import org.apache.flink.api.common.typeinfo.{TypeInfoFactory, TypeInformation, Types}
+import org.apache.flink.api.java.typeutils.TypeExtractor
+
+import java.lang.reflect.Type
+import java.util
+import scala.reflect.{ClassTag, classTag}
+
+object FlinkBaseTypeInfoRegister {
+
+  private val BaseTypes = List(
+    Types.LOCAL_DATE,
+    Types.LOCAL_TIME,
+    Types.LOCAL_DATE_TIME,
+    Types.INSTANT,
+    Types.SQL_DATE,
+    Types.SQL_TIME,
+    Types.SQL_TIMESTAMP,
+  )
+
+  def makeSureBaseTypesAreRegistered(): Unit =
+    BaseTypes.foreach(register(_))
+
+  private def register[T: ClassTag](typeInformation: TypeInformation[T]): Unit = {
+    val klass = classTag[T].runtimeClass.asInstanceOf[Class[T]]
+    val factory = new TypeInfoFactory[T] {
+      override def createTypeInfo(
+          t: Type,
+          genericParameters: util.Map[String, TypeInformation[_]]
+      ): TypeInformation[T] =
+        typeInformation
+    }
+
+    try {
+      TypeExtractor.registerFactory(klass, factory.getClass)
```

Comment:
I can't use there TypeExtractor.get because I've received: 


```
An exception or error caused a run to abort: The class 'pl.touk.nussknacker.engine.flink.api.typeinformation.FlinkBaseTypeInfoRegister$$anon$1' is not instantiable: The class is an inner class, but not statically accessible. 
java.lang.RuntimeException: The class 
```